experiments/react-native-uifabric/src/components/RadioGroup/RadioButton.tsx

```diff
@@ -0,0 +1,122 @@
+/** @jsx withSlots */
+'use strict';
+import * as React from 'react';
+import { View, Text, NativeSyntheticEvent } from 'react-native';
+import {
+  radioButtonName,
+  IRadioButtonType,
+  IRadioButtonProps,
+  IRadioButtonState,
+  IRadioButtonSlotProps,
+  IRadioButtonRenderData
+} from './RadioButton.types';
+import { compose, IUseComposeStyling } from '@uifabricshared/foundation-compose';
+import { filterViewProps } from '../../utilities/RenderHelpers';
+import { ISlots, withSlots } from '@uifabricshared/foundation-composable';
+import { settings } from './RadioButton.settings';
+import { mergeSettings } from '@uifabricshared/foundation-settings';
+import { foregroundColorTokens, textTokens, borderTokens, backgroundColorTokens } from '../../tokens';
+import { useAsPressable } from '../../hooks';
+import { RadioGroupContext } from './RadioGroup';
+
+export const RadioButton = compose<IRadioButtonType>({
+  displayName: radioButtonName,
+
+  usePrepareProps: (userProps: IRadioButtonProps, useStyling: IUseComposeStyling<IRadioButtonType>) => {
+    const { content, buttonKey, disabled, ariaLabel, ...rest } = userProps;
+
+    // Grabs the context information from RadioGroup (currently selected button and client's onChange callback)
+    const info = React.useContext(RadioGroupContext);
+
+    const pressable = useAsPressable({
+      ...rest
+    });
+
+    const state: IRadioButtonState = {
+      ...pressable.state,
+      selected: info.selectedKey === userProps.buttonKey,
+      disabled: disabled || false
+    };
+
+    // Grab the styling information from the userProps, referencing the state as well as the props.
+    const styleProps = useStyling(userProps, (override: string) => state[override] || userProps[override]);
+
+    // This function is called every time a RadioButton gains focus. It does two things:
+    // 1) Calls pressable's onFocus in order to keep track of our state's focus variable. It is dependent on pressable's
+    //    focus variable. Without this, it wouldn't stay updated because we're overriding it's onFocus below for the rootProps.
+    // 2) Selects the currently focused button by calling the RadioGroup's callback function.
+    const onFocusChange = (ev: NativeSyntheticEvent<{}>) => {
```

Comment:
This should probably also be a React.useCallback memoized function

Reply:
Changing now

Reply:
Should it be dependent on [state.focused]?

Reply:
Weird behavior when just doing [state.focused], leaving out of commit for now

Reply:
Fixed in my newest iteration